File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/ASchemaGivenForPrefixitemsTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class ASchemaGivenForPrefixitemsTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.onlyFormat());

    @Test
    public void testCorrectTypesPasses() throws ValidationException {
        // correct types
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        schema.validate(
            new ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

                .add("foo")

            .build(),
            configuration
        );
    }

    @Test
    public void testArrayWithAdditionalItemsPasses() throws ValidationException {
        // array with additional items
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        schema.validate(
            new ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

                .add("foo")

                .add(true)

            .build(),
            configuration
        );
    }

    @Test
    public void testJavascriptPseudoArrayIsValidPasses() throws ValidationException {
        // JavaScript pseudo-array is valid
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        schema.validate(
            MapUtils.makeMap(
                new AbstractMap.SimpleEntry<String, Object>(
                    "0",
                    "invalid"
                ),
                new AbstractMap.SimpleEntry<String, Object>(
                    "1",
                    "valid"
                ),
                new AbstractMap.SimpleEntry<String, Object>(
                    "length",
                    2
                )
            ),
            configuration
        );
    }

    @Test
    public void testEmptyArrayPasses() throws ValidationException {
        // empty array
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        schema.validate(
            new ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
            .build(),
            configuration
        );
    }

    @Test
    public void testWrongTypesFails() {
        // wrong types
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                    "foo",
                    1
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testIncompleteArrayOfItemsPasses() throws ValidationException {
        // incomplete array of items
        final var schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance();
        schema.validate(
            new ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

            .build(),
            configuration
        );
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/ASchemaGivenForPrefixitemsTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class ASchemaGivenForPrefixitemsTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testCorrectTypesPasses() {
        // correct types
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        schema.validate(
            ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

                .add("foo")

            .build(),
            configuration
        )
    }

    @Test
    fun testArrayWithAdditionalItemsPasses() {
        // array with additional items
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        schema.validate(
            ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

                .add("foo")

                .add(true)

            .build(),
            configuration
        )
    }

    @Test
    fun testJavascriptPseudoArrayIsValidPasses() {
        // JavaScript pseudo-array is valid
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "0",
                    "invalid"
                ),
                Pair(
                    "1",
                    "valid"
                ),
                Pair(
                    "length",
                    2
                )
            ),
            configuration
        )
    }

    @Test
    fun testEmptyArrayPasses() {
        // empty array
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        schema.validate(
            ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
            .build(),
            configuration
        )
    }

    @Test
    fun testWrongTypesFails() {
        // wrong types
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf(
                        "foo",
                        1
                    ),
                    configuration
                )
            }
        )
    }

    @Test
    fun testIncompleteArrayOfItemsPasses() {
        // incomplete array of items
        val schema = ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitems1.getInstance()
        schema.validate(
            ASchemaGivenForPrefixitems.ASchemaGivenForPrefixitemsListBuilder()
                .add(1)

            .build(),
            configuration
        )
    }
}
